packages/caravan-clients/src/client.ts

```diff
@@ -240,33 +249,57 @@ export class BlockchainClient extends ClientBase {
   }: BlockchainClientParams) {
     // regtest not supported by public explorers
     if (
-      type !== ClientType.PRIVATE &&
+      type === ClientType.PUBLIC &&
       network !== Network.MAINNET &&
       network !== Network.TESTNET &&
       network !== Network.SIGNET
     ) {
       throw new Error("Invalid network");
     }
-    if (type !== ClientType.MEMPOOL && network === Network.SIGNET) {
-      throw new Error("Invalid network");
+
+    // Blockstream does not support Signet
+    if (type === ClientType.PUBLIC && provider === PublicBitcoinProvider.BLOCKSTREAM && network === Network.SIGNET) {
+      throw new Error("Invalid network: Blockstream does not support Signet");
     }
 
-    let host = "";
+    if (type === ClientType.PRIVATE && provider) {
+      throw new Error("Provider cannot be set for private client type");
+    }
 
-    if (type === ClientType.BLOCKSTREAM) {
-      host = "https://blockstream.info";
-    } else if (type === ClientType.MEMPOOL) {
-      host = "https://unchained.mempool.space";
+    // Backwards compatibility for older configs where client.type = 'mempool' or 'blockstream'
+    if (type === ClientType.MEMPOOL || type === ClientType.BLOCKSTREAM) {
+      // eslint-disable-next-line no-param-reassign
+      provider = type as any;
+      // eslint-disable-next-line no-param-reassign
+      type = ClientType.PUBLIC;
     }
-    if (type !== ClientType.PRIVATE && network !== Network.MAINNET) {
-      host += `/${network}`;
+
+    if (type === ClientType.PUBLIC && !provider) {
+        // Default to blockstream if no provider is specified for a public client
+        // eslint-disable-next-line no-param-reassign
+        provider = PublicBitcoinProvider.BLOCKSTREAM;
     }
-    if (type !== ClientType.PRIVATE) {
-      host += "/api";
+
+    let hostURL = "";
+    if (type === ClientType.PUBLIC) {
+      if (!provider) {
```

Comment:
is this possible? above you check if it's public and if provider is not set, you set it. So this check shouldn't be necessary.

Reply:
Updated.